executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -64,6 +67,35 @@ pub fn split_out_machines<'a, T: FieldElement>(
     let mut base_identities = identities.clone();
     let mut extracted_prover_functions = HashSet::new();
     let mut id_counter = 0;
+
+    let phantom_lookups = identities
+        .iter()
+        .filter_map(|identity| match identity {
+            Identity::PhantomLookup(phantom_lookup) => Some(phantom_lookup.clone()),
+            _ => None,
+        })
+        .collect::<Vec<_>>();
+
+    // Compute mapping: <identity ID> -> <multiplicity column ID>
+    let identity_id_to_multiplicity = phantom_lookups
+        .iter()
+        .map(|id| match id.multiplicity {
+            AlgebraicExpression::Reference(AlgebraicReference { poly_id, .. }) => (id.id, poly_id),
+            _ => {
+                unimplemented!(
+                    "Only simple references are supported, got: {}",
+                    id.multiplicity
+                )
+            }
+        })
+        .collect::<BTreeMap<_, _>>();
+
+    // Multiplicity columns are not connected to any machine, because we removed identities that reference
```

Comment:
I see that the higher-stage polynomial identities involving multiplicity columns are removed, but what about the phantom lookup? Shouldn't the RHS columns include the multiplicity column and thus be connected to the rest of the columns?

So I think we should change `all_row_connected_witnesses` to include the multiplicity column on the RHS columns.

Reply:
No, the multiplicity does not appear in the RHS! It is an extra field in the phantom lookup that is the ignored by the rest of witgen (which treats phantom lookups and native ones the same).

Reply:
But shouldn't we change all_row_connected_witnesses?

Reply:
I added the multiplicities to `lookup_witnesses`, which is the starting point of `all_row_connected_witnesses`. Now, they don't need to be explicitly removed anymore.